icefoganalytics/yhsi
# Issue 18: Change Tracking and Approval: Summary Section Example

# User Story

Build an approval UI for the "Place" fields listed in the [Summary](https://github.com/icefoganalytics/yhsi/blob/test/src/web/src/components/Sites/SitesForms/Summary.vue) form.

Get some feedback and stabilize the format and patterns before allowing the UI to handle the other sub-forms.

# Testing Instructions
1. Update your user roles to include `Site Admin` via the database console.
2. Go to the /sites table and select a site.
3. In the summary form, make and edit and press save.
4. The site will refresh and your edits will be saved.
5.  Update your user roles so that they _do not include_ `Site Admin` or `Administrator` via the database console. Something like `Site Viewer` would probably be best.
6. Go to the /sites table and select a site.
7. In the summary form, make and edit (or two) and press save.
8. The page will refresh and all Summary fields will become readonly. The save button on the Summary form will be disabled with an appropriate tooltip.
9. Make yourself an `Site Admin` again and go to the /sites-change-requests table.
10. The new change request will be at the top of the list. Click on it to review.
11. Note the various features:
   - "accept/reject" toggle buttons
   - accept/reject all
   - save button is locked until all changes are approved/rejected
12. Reject and accept some changes and press Save.
13. Go back to the /sites table and check that you can now edit the site and that the changes have been applied or rejected correctly.
